distributed/scheduler.py

```diff
@@ -807,6 +810,9 @@ def __init__(self, name: str):
         self._nbytes_in_memory = 0
         self._duration = 0
         self._types = set()
+        self._start = 0.0
+        self._stop = 0.0
+        self._all_durations = defaultdict(float)
```

Comment:
Would it be possible to not use `defaultdict`? When using `dict`, we can optimize calls to the object when Cythonizing

Reply:
We could, but it inceases code complexity.  I'm inclined to wait until we see that this is clearly slowing things down before optimizing away from code complexity if that's ok.  I'm enthusiastic to see the C API PR get in.

Reply:
If we do find that this is important we can maybe replace this with a class or namedtuple (maybe)

Reply:
Yeah this is more important with objects seeing a lot of use. Given the limited use of this attribute, this is probably ok for now